File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-1.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int l1=nums.length;
        int min=l1+1,l=0,r=0;
        int s=0;
        while(r<l1)
        {
            s+=nums[r];
            while (s >= target) {
            min = Math.min(min, r - l + 1);
            s -= nums[l];
            l++;
            }
            r++;
            
        }
        if(min==l1+1)
        return 0;
        return min;
    }
}

File: Data Structures & Algorithms/combination-sum-iv/submission-1.py
from functools import cache 
class Solution:
    def combinationSum4(self, nums: List[int], target: int) -> int:
        @cache
        def find(t):
            if t == 0:
                return 1

            if t < 0:
                return 0

            ans = 0

            for num in nums:
                ans += find(t - num)

            return ans

        return find(target)